Vite Migration

kieran-nintex opened the pull request:
- Migrated from Rollup and Babel to Vite
- Migrated from Ava to Vitest
- Updated packages
- Updated readme

Commit:
Upgrade packages and add Vite

Commit:
Upgraded packages and add Vite
- Updated old packages
- Replaced Babel and Rollup with Vite
- Replaced Ava with Vitest
- Renamed testing files to use Vitest defaults

Commit:
Removed unused imports and template test

Commit:
Fixed test naming

kieran-nintex committed:
- Remove unnecessary terser packages
- Update vite config to include .min files
- Created post-process script to minify .min files since vite for es modules does not currently allow minification
- Updated tsconfig to exclude dist and coverage directories
- Update vitest config to only include src directory
- Update README to include ava -> vite migration guide

kieran-nintex committed:
Update config
- Added glob package that was needed by vite config
- Updated README

Signed-off-by: kieran-nintex <kieran.armstrong@nintex.com>

seanhnintex commented on README.md:
```diff
@@ -65,7 +58,225 @@ accessible) and register the service type using the system SmartObject located
 Provider and run the Create From URL method.
 
-### License
+## Migration from Ava to Vitest
```
move the upgrade guide down to the bottom perhaps keep a link at the top.

Add a [[_TOC_]] at the top

kieran-nintex committed:
Update config
- Added glob package that was needed by vite config
- Updated README

Signed-off-by: kieran-nintex <kieran.armstrong@nintex.com>

kieran-nintex committed:
Update readme
- Fixed spelling
- Added ToC
- Moved Migration Guide to bottom

Signed-off-by: kieran-nintex <kieran.armstrong@nintex.com>

kieran-nintex committed:
Update readme
- Added dynamic ToC

Signed-off-by: kieran-nintex <kieran.armstrong@nintex.com>

kieran-nintex committed:
Update readme
- Reverted to generated ToC

Signed-off-by: kieran-nintex <kieran.armstrong@nintex.com>

kieran-nintex committed:
Update readme
- Created separate Migration guide

Signed-off-by: kieran-nintex <kieran.armstrong@nintex.com>

kieran-nintex committed:
Updated tsconfig and vite config
- Got output working with JSSP again
- Removed hacky post-process for minification, replaced with terser plugin
- Cleaned up tsconfig (caused several issues when clashing with vite)

Signed-off-by: kieran-nintex <kieran.armstrong@nintex.com>
